### ISSUE TITLE
AngularFireAuth forces angular to check every second

### ISSUE THREAD

**Issue by Ploppy3** [angular/angularfire2#1037]
### Version info

**Angular:** 4.2.2

**Firebase:** 4.1.2

**AngularFire:** 4.0.0-rc.1

**Other (e.g. Ionic/Cordova, Node, browser, operating system):** Chrome 59

### How to reproduce these conditions

**Steps to set up and reproduce**

Declare AngularFireAuth in any component.

```
constructor( public afAuth: AngularFireAuth, ) {  }

ngDoCheck(){ console.log('check'); }
```

### Expected behavior

AngularFire should not force Angular to do a check every second.

### Actual behavior

AngularFire forces Angular to do a check every second.

**Comment by davideast** [comment]
@Ploppy3 Can you create a plnkr to demonstrate? It makes it much easier for us to troubleshoot.

**Comment by Ploppy3** [comment]
@davideast I don't know how to create a plunker with angular libraries.

**Comment by davideast** [comment]
@Ploppy3 We have this handy dandy plnkr template right here!

http://plnkr.co/edit/8yY4tH?p=preview

**Comment by Ploppy3** [comment]
Click on the url.

**Comment by xtazy** [comment]
Same problem.

**Comment by davideast** [comment]
@xtazy, If you can create a repro that would be great.

**Comment by xtazy** [comment]
I'm not able to reproduce the problem in a plunkr, sorry. It seems to be related to my code, then.

**Comment by Ploppy3** [comment]
@davideast @xtazy same for me, it does not always happen which is really weird. It has nothing to do with my code as explained into the report.

**Comment by xtazy** [comment]
The problem comes from the Firebase library, where a setInterval is registered in the auth.js file. I'll open an issue on their github because I don't remember this was the case before.

**Comment by xtazy** [comment]
Hi there,

Don't know why, but after I restarted my laptop, the problem has gone away.

**Comment by Ploppy3** [comment]
@xtazy As I said the problem only happen sometimes which is extremely weird, you'll probably get it later again.

**Comment by xtazy** [comment]
Indeed, problem has appeared again. In fact, if I close my browser, everything works fine. But as soon as I refresh a page, the setInterval is registered, causing Angular to refresh every second.

Really strange :)

**Comment by xtazy** [comment]
After an intense debug session (:p) I think I found out why the problem appears:

In the auth.js file of firebase there is at some point a check on the User-agent that will later register or not the setInterval. Briefly, if the User-agent matches 'android', then the setInterval is registered, otherwhise not.
The first time I open my browser, the dev tools are not opened and my user agent is the default one -> it doesn't matches android and the setInterval is not registered.
If I open the dev tools in mobile mode, the user agent is modified and 'android' is included in it -> it matches android and the setInterval is registered.

So it seems that there is a different behavior depending if the code is executed on a mobile or not.

I'll try to create plunkr to demonstrate this.

**Comment by xtazy** [comment]
Plunkr created: https://plnkr.co/edit/h6Ku3qtTicNO8N0VEFjU?p=preview
Open a new tab in Chrome, toggle to mobile mode, navigate to the plunk url and then look at the console.

**Comment by jshcrowthe** [comment]
@xtazy thanks for your hard work reproducing this issue! I believe you are correct that it is a JS SDK issue and we can close this in favor of that. Lets give @bojeil-google a chance to verify the issue and we can track this [here](https://github.com/firebase/firebase-js-sdk/issues/119).

**Comment by jamesdaniels** [comment]
Closing for now, feel free to continue discussion here + reopen if we have any action items on this repo.

**Comment by Ploppy3** [comment]
@jamesdaniels Firebase team seems to claim that they use a setInterval and that its Angular's problem.

If a setInterval always trigger a ngDoCheck, should running it outside of zonejs fix the issue?
In this case, could you fix it?

**Comment by jamesdaniels** [comment]
Ah that makes sense, I'll take a peek when I have a few.

**Comment by Ploppy3** [comment]
@jamesdaniels Hello, do you have any information available?

**Comment by jamesdaniels** [comment]
This will be fixed once #1193 is merged and a new version cut, closing for now.

**Comment by Ploppy3** [comment]
@jamesdaniels I know it's been a long time I know but recent events bring me back to this issue. 

I'm wondering if there is still some setInterval/setTimeout running in ngZone. 

Angular recently released their service worker package, and as you can see in this issue https://github.com/angular/angular/issues/20970 I was told that the service worker won't start if the ngZone is not stabilized, the zone never stabilizes if a setInterval/setTimeout is started inside it. So the only solution right now is to run them outside the zone.

In a project where I use only the Angularfire2 & ngx-charts the service worker won't start. So there must be a setInterval/setTimeout running somewhere. I doubt that it is in the ngx-charts library.

Do you know any way to check quickly if all the setInterval/setTimout use in this library are running outside the zone?
